parsers/AR.py

```diff
@@ -20,26 +20,40 @@
 )
 CAMMESA_RENEWABLES_ENDPOINT = "https://cdsrenovables.cammesa.com/exhisto/RenovablesService/GetChartTotalTRDataSource/"
 
+CAMMESA_RENEWABLES_REGIONAL_ENDPOINT = (
+    "https://cdsrenovables.cammesa.com/exhisto/RenovablesService/getCapacidadesRegiones"
+)
+
+CAMMESA_REGIONS_ENDPOINT = "https://api.cammesa.com/demanda-svc/demanda/RegionesDemanda"
+
+REGIONS = {
+    "AR": "Total del SADI ",
+    "AR-BAS": "Buenos Aires",
+    "AR-NEA": "NEA",
+    "AR-NOA": "NOA",
+    "AR-CEN": "Centro",
+    "AR-PAT": "Patagonia",
+    "AR-LIT": "Litoral",
+    "AR-COM": "Comahue",
+    "AR-CUY": "Cuyo",
+}
+
 
 def fetch_production(
     zone_key="AR",
     session: Optional[Session] = None,
     target_datetime: Optional[datetime] = None,
     logger: Logger = getLogger(__name__),
 ) -> List[dict]:
-    """Requests up to date list of production mixes (in MW) of a given country."""
+    """Requests up to date list of production mixes (in MW) of a given region."""
 
     if target_datetime:
         raise NotImplementedError("This parser is not yet able to parse past dates")
 
     current_session = session or Session()
 
-    non_renewables_production: Dict[str, dict] = non_renewables_production_mix(
-        zone_key, current_session
-    )
-    renewables_production: Dict[str, dict] = renewables_production_mix(
-        zone_key, current_session
-    )
+    non_renewables_production = non_renewables_production_mix(zone_key, current_session)
+    renewables_production = renewables_production_mix(zone_key, current_session)
 
     full_production_list = [
         {
```

Comment:
I'm just thinking that here we will miss datapoints which have only non_renewables.
```
renewables = {
'2022-12-22 01:00': {},
'2022-12-22 02:00': {}
}

non_renewables = {
'2022-12-22 01:00': {},
'2022-12-22 03:00': {}
}
Here the 03:00 timestamp would be ignored for non_renewable.
```
Maybe it would be better to rewrite this a bit to merge them if they exist with the same timestamp

Reply:
I would probably add unique keys for hydro (since it exits in both) and then just merge the dicts with the datetime as the key, that would ensure we get all data in one place grouped by the correct datetime and then format the data in a separate function.

Reply:
@unitrium please see my comment here, which addresses your concern:
https://github.com/electricitymaps/electricitymaps-contrib/pull/4894#issuecomment-1364375994

Reply:
Like Robin I am unsure how to best handle this (based on the above comment) as there seem to be no way to ensure the two APIs are in sync.